### DISCUSSION TITLE
Can pinocchio parse xacro files directly

### DISCUSSION THREAD

**Discussion by JiangShangJiu** [stack-of-tasks/pinocchio#2507] [Q&A] [answered]
Can pinocchio parse xacro files directly

**Comment by jorisv** [comment]
Hello @JiangShangJiu,

No, pinocchio can parse URDF, but it can't parse xacro macro language.

**Comment by jcarpent** [comment, marked as answer]
You can translate xacro into URDF easily: https://docs.unity3d.com/Simulation/manual/author/create-a-vehicle-model/convert-a-xacro-file.html
